src/geoapis/lidar.py

```diff
@@ -205,56 +208,79 @@ def _get_dataset_tile_names(self, client, dataset_prefix):
 
         return tile_info
 
-    def _calculate_dataset_download_size(self, client, dataset_prefix, tile_info):
+    def _calculate_dataset_download_size(self, client, _dataset_prefix, tile_info):
         """Sum up the size of the LiDAR data in catchment"""
         lidar_size_bytes = 0
 
         for tile_url in tile_info.urls:
-            # drop the OT_BUCKET from the URL path to get the file_prefix
-            file_prefix = pathlib.Path(
+            # drop the OT_BUCKET from the URL path to get the file_name
+            file_name = pathlib.Path(
                 *pathlib.Path(urllib.parse.urlparse(tile_url).path).parts[2:]
             )
-            local_path = self.cache_path / file_prefix
+            local_path = self.cache_path / file_name
             if self.redownload_files_bool or not local_path.exists():
 
                 try:
                     response = client.head_object(
-                        Bucket=self.OT_BUCKET, Key=str(file_prefix.as_posix())
+                        Bucket=self.OT_BUCKET, Key=str(file_name.as_posix())
                     )
                     lidar_size_bytes += response["ContentLength"]
 
                     if self.verbose:
                         print(
-                            f"checking size: {file_prefix}: {response['ContentLength']}"
+                            f"checking size: {file_name}: {response['ContentLength']}"
                             f", total (GB): {self._to_gbytes(lidar_size_bytes)}"
                         )
                 except botocore.exceptions.ClientError as e:
-                    f"An error occured during access of {file_prefix}, The error is {e}"
+                    f"An error occured during access of {file_name}, The error is {e}"
         return lidar_size_bytes
 
-    def _download_tiles_in_catchment(self, client, dataset_prefix, tile_info):
+    def _download_tiles_in_catchment(
+        self, client, dataset_prefix, tile_info, lidar_size_bytes
+    ):
         """Download the LiDAR data within the catchment"""
+        total_gb = self._to_gbytes(lidar_size_bytes)
+        # Create progress bar context, only displayed if self.verbose is True
+        with tqdm(
+            disable=not self.verbose, unit="GB", total=total_gb, ncols=140
```

Comment:
@LukeParky I think this is a really nice improvement in the user experience.

I'm working about the impact on the dependency footprint in adding tqdm? I'm thinking if it's minimal (i.e. a base library generally included anyway) we just go ahead and bring this back into the base geoapis. If it doesn't play nice with geopandas/rioxarray I think we introduce a [verbose] vs [standard] option. This second approach will be much more complicated now that I write it so hopefully we can just go with the first option.

How does meeting Wednesday to discuss work? I'll make a little reminder meeting :)

Reply:
happy to not worry about this!